fastlane/lib/fastlane/plugins/plugin_info_collector.rb

```diff
@@ -49,7 +55,20 @@ def plugin_name_valid?(name)
       /^[a-z0-9_]+$/ =~ name &&
         # Does not contain the words 'fastlane' or 'plugin' since those will become
         # part of the gem name
-        [/fastlane/, /plugin/].none? { |regex| regex =~ name }
+        [/fastlane/, /plugin/].none? { |regex| regex =~ name } &&
+        # Plugin name isn't taken on RubyGems yet
+        !plugin_name_taken?(name)
+    end
+
+    # Checks if the plugin name is still free on RubyGems
+    def plugin_name_taken?(name)
+      require 'open-uri'
+      require 'json'
+      url = "https://rubygems.org/api/v1/gems/#{name}.json"
+      response = JSON.parse(open(url).read)
+      return !!response['version']
+    rescue
+      false
```

Comment:
Yeah, but that would be a cache that contains a hash, since the parameter is different every time. I  figured since this runs only when creating a new plugin, and speed is not super important, it's okay to send the request twice instead of adding caching complexity to this method. In general the RubyGems API is really fast.